### DISCUSSION TITLE
Help Needed with iPLASS Authentication and Authorization

### DISCUSSION THREAD

**Discussion by khanh7516** [dentsusoken/iPLAss#1551] [Q&A] [answered]
I'm working on adding user authentication and authorization to my application using the iPLAss platform, and I've hit a few roadblocks.

### Challenges:

* I'm having trouble figuring out the right approach to implement the authentication and authorization logic with iPLAss.
* I’ve gone through the [認証](https://iplass.org/en/docs/developerguide/authentication/index.html), but some parts are still unclear to me.
* I found the web APIs for authentication and authorization, but I'm not sure how to use them effectively.
### What I Need:
I'd really appreciate it if someone could share a clear example of how to set up authentication and authorization with iPLAss. A step-by-step guide or sample code would be incredibly helpful.

Thanks so much for your help!

**Comment by KoheiKGYM** [comment, marked as answer]
Hi @khanh7516,
we appreciate your use of iPLAss.

We understand that you want to call the authentication process of iPLAss from your application using Web API. If you try to do this process, please follow the provided steps.

1. Create your custom Web API for authentication
    - Please refer to the following link to create the custom Web API.
    https://iplass.org/docs/developerguide/webapi/index.html#_%E3%82%AB%E3%82%B9%E3%82%BF%E3%83%A0_webapi
    https://iplass.org/docs/developerguide/customizing/index.html#Command
    - In your custom Web API, implement the authentication process using Java API of iPLAss. You can use the login() method of AuthManager of iPLAss as called in the link below.
    https://github.com/dentsusoken/iPLAss/blob/master/iplass-gem/src/main/java/org/iplass/gem/command/auth/LoginCommand.java
    - Example of the Groovy command called in Web API
      ```
      import org.iplass.mtp.ManagerLocator
      import org.iplass.mtp.auth.AuthManager
      import org.iplass.mtp.auth.login.IdPasswordCredential

      String id = request.getParam("PARAM_USER_ID");
      String pass = request.getParam("PARAM_PASSWORD");

      AuthManager authManager = ManagerLocator.getInstance().getManager(AuthManager.class)
      IdPasswordCredential cre = new IdPasswordCredential(id, pass)
      authManager.login(cre)
      return "SUCCESS";
2. Once authentication is completed, a session cookie (JSESSIONID) is issued. You can keep the login state for subsequent requests by sending this cookie.

**Reply by khanh7516** [reply]
Hi @KoheiKGYM,
thank you so much for your help. Your guidance made a big difference and really helped me out.
